commandapi-annotations-reloaded-tests/src/main/java/dev/jorel/commandapi/annotations/reloaded/test/AnnotatedCommandBossbar.java

```diff
@@ -0,0 +1,203 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *******************************************************************************/
+package dev.jorel.commandapi.annotations.reloaded.test;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import dev.jorel.commandapi.annotations.reloaded.annotations.Executes;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.NamespacedKey;
+import org.bukkit.boss.BarColor;
+import org.bukkit.boss.BarStyle;
+import org.bukkit.boss.BossBar;
+import org.bukkit.boss.KeyedBossBar;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import dev.jorel.commandapi.CommandAPI;
+import dev.jorel.commandapi.annotations.reloaded.annotations.Command;
+import dev.jorel.commandapi.annotations.reloaded.annotations.Subcommand;
+import dev.jorel.commandapi.annotations.reloaded.arguments.ABooleanArgument;
+import dev.jorel.commandapi.annotations.reloaded.arguments.AChatComponentArgument;
+import dev.jorel.commandapi.annotations.reloaded.arguments.AEntitySelectorArgument;
+import dev.jorel.commandapi.annotations.reloaded.arguments.AIntegerArgument;
+import dev.jorel.commandapi.annotations.reloaded.arguments.AMultiLiteralArgument;
+import dev.jorel.commandapi.annotations.reloaded.arguments.ANamespacedKeyArgument;
+import dev.jorel.commandapi.exceptions.WrapperCommandSyntaxException;
+import net.md_5.bungee.api.chat.BaseComponent;
+
+/**
+ * The annotated equivalent of BetterBossBars from the
+ * examples/bukkit/commandtrees directory
+ */
+@Command("betterbossbar")
+public class AnnotatedCommandBossbar {
+
+	// /betterbossbar set <id> players <targets>
+	// /betterbossbar set <id> style (notched_6|notched_10|notched_12|notched_20|progress)
+	// /betterbossbar set <id> value <value>
+	// /betterbossbar set <id> visible <visible>
+
+	// /betterbossbar remove <id>
+
+	// /betterbossbar get <id> players
+	// /betterbossbar get <id> visible
+	// /betterbossbar get <id> max
+	// /betterbossbar get <id> value
+
+	// /betterbossbar add <id> <name>
+
+	// /betterbossbar list
+
+	private Map<NamespacedKey, Integer> maxValues;
+
+	public AnnotatedCommandBossbar() {
+		this.maxValues = new HashMap<>();
+	}
+
+	@Subcommand("set")
+	class SetSubcommand {
+
+		@ANamespacedKeyArgument
+		NamespacedKey id;
+
+		@Subcommand("players")
+		public void players(CommandSender sender,
```

Comment:
From reading the spec doc, I'm not entirely clear on the purpose of `@Executes` over `@Subcommand` or why both would need to be included. What does this actually mean and how should it affect the generated code?

Reply:
Well, `@Executes` is supposed to mark a method as executable.
The usage of `@Subcommand` can change a bit. You can either use this multiple times on an inner class to not create too many nested classes, or you actually do all those nested class and put one `@Subcommand` per inner class.
Or, in cases like this one, you can omit the inner class and put the `@Subcommand` annotation on the method - alongside `@Executes`

Reply:
Thinking about what is nicest for the user then, I think the `@Executes` on this method can be inferred from the presence of `@Subcommand`. The intention is clear, so it'd be better if this just works rather than requiring them to add an extra annotation to make it work.

Reply:
Yeah, but I don't think it should be that way since the way the `@Subcommand` annotation is used here is syntactic sugar for this
```java
@Command
public class Blah {
    @Subcommand
    public class BlahSub {
        @Subcommand
        @Executes
        public void executorMethod(Player player) {}
    }
}
```

instead of this:
```java
@Command
public class Blah {
    @Subcommand
    public class BlahSub {
        @Subcommand
        public class BlahSubSub {
            @Executes
            public void executorMethod(Player player) {}
        }
    }
}
```

Reply:
I feel like I am missing something. Yes, it's syntactic sugar. I don't see why that means we should make the user add an extra annotation that can be inferred. What problem does this cause?

Reply:
No, my point is that because it's syntactic sugar we should require the extra `@Executes` annotation.
I don't know, maybe that's just my personal preference or maybe I just want to stick to the spec.
Maybe we just need other opinions. I think I've stated mine.